### PR TITLE
[x86/Linux] 16-byte aligned HelperMethodFrameRestoreState

### PR DESCRIPTION
**Pull request by parjong**
This commit revises HelperMethodFrameRestoreState to emit 16-byte aligned frame before invoking HelperMethodFrameConfirmState.

### CODE REVIEW

**Comment by parjong** [comment]
This PR is currently under internal evaluation.

**Comment by parjong** [comment]
Internal evaluation confirms that this commit introduces no regression.

**Comment by parjong** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test please

**Comment by parjong** [comment]
@janvorli Please take a look :)

File: src/vm/i386/asmhelpers.S [hunk 1]
```diff
@@ -359,12 +359,28 @@ LEAF_ENTRY HelperMethodFrameRestoreState, _TEXT
...
+    CHECK_STACK_ALIGNMENT
     call        C_FUNC(HelperMethodFrameConfirmState)
+    add         esp, 4
+
+    // Restore EBP
+    EPILOG_BEG
+    EPILOG_END
+
     // on return, eax = MachState*
     cmp         DWORD PTR [eax + MachState__pRetAddr], 0
```
Comment (janvorli) [inline src/vm/i386/asmhelpers.S]: After your change, EAX is no longer pointing to MachState here.

File: src/vm/i386/asmhelpers.S [hunk 2]
```diff
@@ -359,12 +359,28 @@ LEAF_ENTRY HelperMethodFrameRestoreState, _TEXT
...
-    push        ebp
+
+    mov         eax, ebp
+
+    // Create a minimal EBP-frame (for clean stack trace under debugger)
+    PROLOG_BEG
+    PROLOG_END
+
+    // Call HelperMethodFrameConfirmState
+    sub         esp, 4
```
Comment (janvorli) [inline src/vm/i386/asmhelpers.S]: Could you please add comment that this is to make the stack aligned?

File: src/vm/i386/asmhelpers.S (hunk 1 above)
Comment (parjong) [inline src/vm/i386/asmhelpers.S]: It seems that ``eax`` points to ``MachState`` as ``HelperMethodFrameConfirmState`` returns  a pointer to ``MachState`` .

Comment (janvorli) [inline src/vm/i386/asmhelpers.S]: Ah, right, I've missed that.

**Comment by parjong** [comment]
@dotnet-bot test Tizen armel Cross Release Build please.

**Comment by parjong** [comment]
@janvorli CI is all green :)

**Pull request merged**